### DISCUSSION TITLE
Fundamento de Orientação a Objetos - Interfaces

### DISCUSSION THREAD

**Discussion by acgotola** [balta-io/forum#707] [Tira Dúvidas] [answered]
Fiquei com uma dúvida com relação a Herança e Interface.

Interfaces são os contratos e obriga a implementação e pelo que estou entendendo isso acaba gerando polimorfismo.

Já a Herança eu estou herdando propriedades e métodos da classe pai e posso ou não ter polimorfismo.

Estaria correto pensar assim?

Fiquei com dúvida quando escolher um ou outro.

**Comment by JonatasSilva01** [comment, marked as answer]
Nas minhas anotações de c# - Orientação a objetos, Herança é ato de herdar propriedades, métodos ou eventos de uma base deis que esses "propriedades, métodos ou eventos" sejam públicos ou protegidos (public, protected). Em resumo serve para reaproveitar código.

Interfaces servem para definir contratos de implementação. Elas especificam os métodos e propriedades que uma classe deve implementar, garantindo que a classe siga uma estrutura definida, sem escapar das implementações esperadas
